chips/apollo3/src/iom.rs

```diff
@@ -1230,8 +1230,13 @@ impl<'a> SpiMaster<'a> for Iom<'a> {
             Option<SubSliceMut<'static, u8>>,
         ),
     > {
-        let write_len = write_buffer.len();
-        let read_len = read_buffer.as_ref().map_or(0, |b| b.len());
+        let mut write_len = write_buffer.len();
+        let mut read_len = read_buffer.as_ref().map_or(0, |b| b.len());
+
+        if read_len > 0 {
+            write_len = write_len.min(read_len);
+            read_len = write_len;
+        }
```

Comment:
```suggestion
        let (write_len, read_len) = if let Some(rb) = read_buffer.as_ref() {
            let min = write_buffer.len().min(rb.len());
            (min, min)
        } else {
            (write_buffer.len(), 0)
        };
```

Happy to be told to buzz off if this is more confusing than useful, but in my continuing quest to be a better Rust programmer, I think this would be a bit more idiomatic and avoid mutable length variables?